ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/cdr/repositories/CdrsEmspRepository.kt

```diff
@@ -0,0 +1,43 @@
+package com.izivia.ocpi.toolkit.modules.cdr.repositories
+
+import com.izivia.ocpi.toolkit.common.URL
+import com.izivia.ocpi.toolkit.modules.cdr.domain.Cdr
+
+/**
+ * Typically implemented by market roles like: eMSP.
+ *
+ * The CDRs endpoint can be used to create and retrieve CDRs.
+ *
+ * - GET: Retrieve an existing CDR.
+ * - POST: Send a new CDR.
+ * - PUT: n/a (CDRs cannot be replaced)
+ * - PATCH: n/a (CDRs cannot be updated)
+ * - DELETE: n/a (CDRs cannot be removed)
+ */
+interface CdrsEmspRepository<T> {
+    /**
+     * GET Method
+     *
+     * Fetch CDRs from the receivers' system.
+     *
+     * Endpoint structure definition:
+     *
+     * No structure defined. This is open to the eMSP to define, the URL is provided to the CPO by the eMSP in
+     * the result of the POST request. Therefore, OCPI does not define variables.
+     *
+     * @return Cdr Requested Cdr object.
+     */
+    suspend fun getCdr(param: T): Cdr?
```

Comment:
this does not have to use generics, because it's the interface that will be implemented by the user of the lib ;)

```suggestion
interface CdrsEmspRepository {
    /**
     * GET Method
     *
     * Fetch CDRs from the receivers' system.
     *
     * Endpoint structure definition:
     *
     * No structure defined. This is open to the eMSP to define, the URL is provided to the CPO by the eMSP in
     * the result of the POST request. Therefore, OCPI does not define variables.
     *
     * @return Cdr Requested Cdr object.
     */
    suspend fun getCdr(id: String): Cdr?
```